src/Core/Types.Filters/Extensions/FilterFieldCollectionExtensions.cs

```diff
@@ -0,0 +1,83 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+
+namespace HotChocolate.Types.Filters.Extensions
+{
+    public static class FilterFieldCollectionExtensions
+    {
+        public static T GetOrAddDescriptor<T>(
+            this IList<FilterFieldDescriptorBase> fields,
+            PropertyInfo propertyInfo,
+            Func<T> valueFactory) where T : FilterFieldDescriptorBase
+        {
+            if (fields == null)
+            {
+                throw new ArgumentNullException(nameof(fields));
+            }
+            if (propertyInfo == null)
+            {
+                throw new ArgumentNullException(nameof(propertyInfo));
+            }
+            if (valueFactory == null)
+            {
+                throw new ArgumentNullException(nameof(valueFactory));
+            }
+
+            FilterFieldDescriptorBase fieldDescriptor =
+                fields.FirstOrDefault(t => t.Definition.Property.Equals(propertyInfo));
+
+            if (fieldDescriptor is { })
```

Comment:
Shouldn't this lead to a schema exception? How would it be possible to have a different descriptor for the same property on the same filter type?

Reply:
The problem solved  here is this one. 

https://github.com/ChilliCream/hotchocolate/blob/master/src/Core/Types.Filters/IgnoredFilterFieldDescriptor.cs

```
descirptor =>  descriptor.Ignore(x => x.StringField); 
...

descirptor =>  descriptor.Filter(x => x.StringField);  // Exception
```

Reply:
done